### PR TITLE
refactor(router): use strict equality

### PR DESCRIPTION
**Pull request by michaelchiche**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

Does not change

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

use strict equality and not coercion
since the first check verifies that this.target is a string, we can use strict equality to this.target

### CODE REVIEW

File: modules/@angular/router/src/directives/router_link.ts [hunk 1]
```diff
@@ -141,7 +141,7 @@ export class RouterLinkWithHref implements OnChanges, OnDestroy {
       return true;
     }
 
-    if (typeof this.target === 'string' && this.target != '_self') {
+    if (typeof this.target === 'string' && this.target !== '_self') {
```
Comment (vicb) [inline modules/@angular/router/src/directives/router_link.ts]: then the first `typeof` becomes useless right ? should be `if (this.target !== '_self')`

Comment (michaelchiche) [inline modules/@angular/router/src/directives/router_link.ts]: true ;)

Comment (astec) [inline modules/@angular/router/src/directives/router_link.ts]: Why not to combine two `if`s in one?
```
     if (button !== 0 || ctrlKey || metaKey || this.target !== '_self') {
        return true;
      }
```
Does not affect readability & few bytes gained.

Comment (michaelchiche) [inline modules/@angular/router/src/directives/router_link.ts]: added those modifications

**Comment by vicb** [comment]
please squash your commit and update the msg to `refactor(router): ...` there is no fix in this PR

**Comment by michaelchiche** [comment]
done

**Comment by astec** [comment]
@michaelchiche I guess Victor was talking about commit message when asked to renamed to "refactor(router): ..."

**Comment by vicb** [comment]
Seems like some tests are failing, could you please investigate why ?

**Comment by vicb** [comment]
@michaelchiche any update, otherwise I'll close this PR

**Comment by michaelchiche** [comment]
@vicb I tested on my computer. The tests pass...

**Comment by vicb** [comment]
I have restarted the tests, let's see

**Comment by michaelchiche** [comment]
@vicb  well the tests seem to fail on travis. I do not understand why, since the behavior is the same, its just a small refactor. all tests pass on my laptop. so if it's not good for you, you can close this PR.

**Pull request closed without merging**